### ISSUE TITLE
Add AST AudioSet classification model package

### ISSUE THREAD

**Issue by luisquintanilla** [luisquintanilla/model-packages-prototype#36]
## Summary
Create a sample model package for Audio Spectrogram Transformer (AST) fine-tuned on AudioSet  audio classification across 527 categories.

>  **ONNX Status:** Pre-exported at [`onnx-community/ast-finetuned-audioset-10-10-0.4593-ONNX`](https://huggingface.co/onnx-community/ast-finetuned-audioset-10-10-0.4593-ONNX)  **no manual export needed.**

## HuggingFace Source
- **Original:** [`MIT/ast-finetuned-audioset-10-10-0.4593`](https://huggingface.co/MIT/ast-finetuned-audioset-10-10-0.4593) (PyTorch)
- **Pre-exported ONNX:** [`onnx-community/ast-finetuned-audioset-10-10-0.4593-ONNX`](https://huggingface.co/onnx-community/ast-finetuned-audioset-10-10-0.4593-ONNX) 

 **Note the `-ONNX` suffix** in the onnx-community repo name.

## Files for Manifest

| File | Manifest Path | Approx Size | Purpose |
|------|--------------|-------------|---------|
| Model | `onnx/model.onnx` | ~345 MB | ViT-based audio spectrogram  527 AudioSet labels |

#### `SampleModelPackage.AstAudioSet.csproj`

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net10.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <Description>Sample model package  MIT/ast-finetuned-audioset (audio classification via ONNX).</Description>
  </PropertyGroup>
  <ItemGroup>
    <ProjectReference Include="..\..\src\ModelPackages\ModelPackages.csproj" />
    <PackageReference Include="MLNet.AudioInference.Onnx" Version="0.1.0-preview.1" />
  </ItemGroup>
  <ItemGroup>
    <EmbeddedResource Include="model-manifest.json" />
  </ItemGroup>
  <ItemGroup>
    <None Include="build\SampleModelPackage.AstAudioSet.props" Pack="true" PackagePath="build\" />
    <None Include="build\SampleModelPackage.AstAudioSet.targets" Pack="true" PackagePath="build\" />
  </ItemGroup>
</Project>
```

#### `model-manifest.json`

```json
{
  "model": {
    "id": "MIT/ast-finetuned-audioset-10-10-0.4593",
    "revision": "main",
    "files": [
      { "path": "onnx/model.onnx", "sha256": "<compute-after-download>", "size": 0 }
    ]
  },
  "sources": {
    "huggingface": {
      "type": "huggingface",
      "repo": "onnx-community/ast-finetuned-audioset-10-10-0.4593-ONNX",
      "revision": "main"
    }
  },
  "defaultSource": "huggingface"
}
```

#### `AstAudioSetModel.cs`

```csharp
using Microsoft.ML;
using MLNet.AudioInference.Onnx;
using ModelPackages;

namespace SampleModelPackage.AstAudioSet;

public static class AstAudioSetModel
{
    // 527 AudioSet labels  sourced from config.json id2label in the HuggingFace repo
    // See: https://huggingface.co/MIT/ast-finetuned-audioset-10-10-0.4593/blob/main/config.json
    private static readonly string[] AudioSetLabels = LoadAudioSetLabels();

    private static readonly Lazy<ModelPackage> Package = new(() =>
        ModelPackage.FromManifestResource(typeof(AstAudioSetModel).Assembly));

    public static Task<ModelFiles> EnsureFilesAsync(
        ModelOptions? options = null, CancellationToken ct = default)
        => Package.Value.EnsureFilesAsync(options, ct);

    public static async Task<OnnxAudioClassificationTransformer> CreateClassifierAsync(
        MLContext? mlContext = null,
        ModelOptions? options = null,
        CancellationToken ct = default)
    {
        var files = await Package.Value.EnsureFilesAsync(options, ct);
        mlContext ??= new MLContext();

        var estimator = new OnnxAudioClassificationEstimator(mlContext,
            new OnnxAudioClassificationOptions
            {
                ModelPath = files.PrimaryModelPath,
                FeatureExtractor = new MelSpectrogramExtractor(16000) { NumMelBins = 128 },
                Labels = AudioSetLabels,
                SampleRate = 16000
            });

        var dummyData = mlContext.Data.LoadFromEnumerable(Array.Empty<AudioInput>());
        return (OnnxAudioClassificationTransformer)estimator.Fit(dummyData);
    }

    private static string[] LoadAudioSetLabels()
    {
        // TODO: Embed the 527 AudioSet label strings from config.json
        // Download from: https://huggingface.co/MIT/ast-finetuned-audioset-10-10-0.4593/raw/main/config.json
        // Parse "id2label"  string[] ordered by integer key
        throw new NotImplementedException("Embed AudioSet labels from config.json id2label mapping");
    }

    private sealed class AudioInput { public float[] Audio { get; set; } = []; }
}
```

#### `SampleConsumer.AstAudioSet.csproj`

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net10.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>
  <ItemGroup>
    <ProjectReference Include="..\SampleModelPackage.AstAudioSet\SampleModelPackage.AstAudioSet.csproj" />
  </ItemGroup>
</Project>
```

#### `Program.cs`

```csharp
using Microsoft.ML;
using MLNet.Audio.Core;
using SampleModelPackage.AstAudioSet;
using System.Diagnostics;

Console.WriteLine("=== AST AudioSet Audio Classification Demo ===\n");

var sw = Stopwatch.StartNew();
var mlContext = new MLContext();
var classifier = await AstAudioSetModel.CreateClassifierAsync(mlContext: mlContext);
Console.WriteLine($"Model ready in {sw.ElapsedMilliseconds}ms\n");

// Load a WAV file (16kHz mono)
var audio = AudioIO.LoadWav("test.wav");
var data = mlContext.Data.LoadFromEnumerable(new[] { new { Audio = audio.Samples } });
var result = classifier.Transform(data);

var predictions = mlContext.Data.CreateEnumerable<ClassificationOutput>(result, reuseRowObject: false);
foreach (var p in predictions)
    Console.WriteLine($"Top prediction: {p.PredictedLabel} (score: {p.Score:F4})");

record ClassificationOutput
{
    public string PredictedLabel { get; set; } = "";
    public float Score { get; set; }
}
```

### Solution Entry

```xml
<Folder Name="/samples/ast-audioset/">
  <Project Path="samples\SampleModelPackage.AstAudioSet\SampleModelPackage.AstAudioSet.csproj" />
  <Project Path="samples\SampleConsumer.AstAudioSet\SampleConsumer.AstAudioSet.csproj" />
</Folder>
```

## OnnxAudioClassificationOptions Reference

- `ModelPath` (required)  path to `model.onnx`
- `FeatureExtractor` (required)  instance of `IFeatureExtractor` (use `MelSpectrogramExtractor(16000) { NumMelBins = 128 }`)
- `Labels` (required)  `string[]` of 527 AudioSet category labels
- `SampleRate`  default 16000

## Model Architecture

| Aspect | Value |
|--------|-------|
| Architecture | Audio Spectrogram Transformer (ViT-based) |
| Parameters | 87M |
| Labels | 527 AudioSet categories |
| Input | 16kHz audio  128-bin mel spectrogram |
| Training | Finetuned on AudioSet 10-10-0.4593 |

## Implementation Notes

- The 527 AudioSet labels must be extracted from the `config.json` `id2label` mapping in the HF repo
- Consider embedding labels as a static array or as a `.json` embedded resource
- AST uses 128 mel bins (not 80) with 16kHz sample rate

## Steps

1. Download ONNX: `huggingface-cli download onnx-community/ast-finetuned-audioset-10-10-0.4593-ONNX --include "onnx/model.onnx" --local-dir models/ast`
2. Extract 527 labels from `config.json` `id2label` mapping
3. Compute SHA256 hash, update manifest
4. Create all files
5. Test via consumer

## Model Garden Migration

Once validated, create `DotnetAILab.ModelGarden.AudioClassification.AstAudioSet` in `dotnet-model-garden-prototype` under `models/audio-classification/`.

## Acceptance Criteria

- [ ] Manifest has correct SHA256 hash
- [ ] `AstAudioSetModel.CreateClassifierAsync()` works
- [ ] Consumer classifies audio into AudioSet categories
- [ ] All 527 labels are correctly embedded

## Dependencies
- `MLNet.AudioInference.Onnx` from [mlnet-audio-custom-transforms](https://github.com/luisquintanilla/mlnet-audio-custom-transforms)
- `ModelPackages` from this repo (ProjectReference)
